cdap-security/src/main/java/co/cask/cdap/security/server/ExternalAuthenticationServer.java

```diff
@@ -214,8 +194,40 @@ protected void startUp() {
 
       server.setHandler(handlers);
     } catch (Exception e) {
-      LOG.error("Error while starting server.", e);
+      LOG.error("Error while starting Authentication Server.", e);
+    }
+
+    try {
+      server.start();
+    } catch (Exception e) {
+      Throwable cause = Throwables.getRootCause(e);
+      if (!(cause instanceof BindException)) {
+        throw e;
+      }
+      throw new ServiceBindFailure("Authentication Server", bindAddress.getCanonicalHostName(), port);
     }
+
+    // assumes we only have one connector
+    final Connector connector = server.getConnectors()[0];
```

Comment:
how come we're moving this logic to startUp?  If the run method is empty should we just use an AbstractIdleService?